### PR TITLE
Get ready for Journal of Open Source Software publication

### PR DESCRIPTION
**Pull request by CJ-Wright**
See here: http://joss.theoj.org/about#author_guidelines

### CODE REVIEW

**Comment by CJ-Wright** [comment]
@mrocklin @ordirules @danielballan @sbillinge Please look this over and either push to this branch or PR into this branch any changes. (Or just email me)

**Comment by sbillinge** [comment]
Thanks CJ

Please could you refer to the paper submission checklist for the group on group google groups and go through it before anything is submitted.  Thanks.

File: paper/paper.md [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+   orcid: 0000-0003-2522-7028
+   affiliation: 2
+ - name: Julien Lhermitte
+   orcid: 0000-0003-0660-975X
+   affiliation: 5
+ - name: Daniel B. Allan
+   orcid: ?
+   affiliation: 4
+ - name: Simon J. L. Billinge
+   orcid: ?
```
Comment (sbillinge) [inline paper/paper.md]: Simon ORCID: 0000-0002-9734-4998

File: paper/paper.md [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+   affiliation: 2, 3
+affiliations:
+ - name: Continuum Analytics
+   index: 1
+ - name: Department of Applied Physics and Applied Mathematics, Columbia University
```
Comment (sbillinge) [inline paper/paper.md]: usually there is a zip code or sthg for affiliation.  Please check journal style.

File: paper/paper.md [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+   affiliation: 2
+ - name: Julien Lhermitte
+   orcid: 0000-0003-0660-975X
+   affiliation: 5
+ - name: Daniel B. Allan
+   orcid: ?
+   affiliation: 4
+ - name: Simon J. L. Billinge
+   orcid: ?
+   affiliation: 2, 3
```
Comment (sbillinge) [inline paper/paper.md]: Simon not part of NSLS-II

Comment (CJ-Wright) [reply paper/paper.md]: That is number 4

File: paper/paper.md (hunk 2 above)
Comment (CJ-Wright) [reply paper/paper.md]: It seems that most don't have a zip code.

File: paper/paper.md [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+Citations to entries in paper.bib should be in
+[rMarkdown](http://rmarkdown.rstudio.com/authoring_bibliographies_and_citations.html)
+format.
+
+# References
+
+# Acknowledgments
+Software and writing contributions from Wright and Billinge were supported by the U.S. National Science Foundation through
+grant DMREF-1534910
+This research used resources of the Center for Functional Nanomaterials, which is a U.S. DOE Office of Science Facility, at Brookhaven National Laboratory under Contract No. DE-SC0012704
```
Comment (ordirules) [inline paper/paper.md]: Just to add after discussion with @stuartcampbell, this should include NSLS-II (same contract number):

"This research used resources of the Center for Functional Nanomaterials and the NSLS-II, which is a U.S. DOE Office of Science Facility, at Brookhaven National Laboratory under Contract No. DE-SC0012704"

File: paper/paper.md [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+   orcid: 0000-0002-9734-4998
+   affiliation: 2, 3
+affiliations:
+ - name: Continuum Analytics
+   index: 1
+ - name: Department of Applied Physics and Applied Mathematics, Columbia University
+   index: 2
+ - name: Condensed Matter Physics and Materials Science Department, Brookhaven National Laboratory
+   index: 3
+ - name: NSLS-II, Brookhaven National Laboratory
```
Comment (ordirules) [inline paper/paper.md]: @danielballan should this be spelled out in full or abbreviated?

**Comment by stuartcampbell** [comment]
As Kevin and Simon are listed, would it be appropriate to add myself ?

**Comment by CJ-Wright** [comment]
Sure, can I please have your ORCID?

**Comment by stuartcampbell** [comment]
Thank you - 0000-0001-7079-0878

File: paper/paper.md (hunk 5 above)
Comment (danielballan) [reply paper/paper.md]: I think NSLS-II works fine here, given that Brookhaven is spelled out.

File: paper/paper.md [hunk 6]
```diff
@@ -0,0 +1,58 @@
...
+ - name: Center for Functional Nanomaterials, Brookhaven National Laboratory
+   index: 5
+date: 19 September 2017
+bibliography: paper.bib
+---
+
+# Summary
+
+- This is a small library to manage continuous streams of data, particularly when complex branching and control flow situations arise. 
+- This provides a framework for streaming data analysis in pure python with hooks into the Dask task scheduler for automatic parallelization.
```
Comment (danielballan) [inline paper/paper.md]: python -> Python

File: paper/paper.md [hunk 7]
```diff
@@ -0,0 +1,58 @@
...
+- Streamz is similar to reactive
+programming systems like [RxPY](https://github.com/ReactiveX/RxPY) or big
+data streaming systems like [Apache Flink](https://flink.apache.org), [Apache Beam](https://beam.apache.org/get-started/quickstart-py) or [Apache Spark Streaming](https://spark.apache.org/streaming/).
+- This software forms the backbone for data processing at the NSLS-II x-ray powder diffraction and complex materials scattering data analysis pipelines.
```
Comment (danielballan) [inline paper/paper.md]: I usually see X-ray with a capital X.

**Comment by CJ-Wright** [comment]
@mrocklin Can I get your ORCID? (Or can you push it to this branch?)

**Comment by CJ-Wright** [comment]
Things left to do (I think, feel free to add more):
- [ ] Turn all the references (spark, flink, etc.) into real references like https://github.com/arfon/fidgit/blob/master/paper/paper.md
- [ ] Double check that all funding and contracts are acknowledged
- [ ] Discuss paper length and content. (we may want to link to the blog post at the very least?)

File: paper/paper.md [hunk 8]
```diff
@@ -0,0 +1,57 @@
...
+---
+
+# Summary
+
+- This is a small library to manage continuous streams of data, particularly when complex branching and control flow situations arise. 
+- This provides a framework for streaming data analysis in pure Python with hooks into the Dask task scheduler for automatic parallelization.
+- Streamz is similar to reactive
+programming systems like RxPY [@RxPy] or big
+data streaming systems like Apache Flink [@Flink], Apache Beam [@Beam] or Apache Spark Streaming [@Spark].
+- This software forms the backbone for data processing at the NSLS-II X-ray powder diffraction and complex materials scattering data analysis pipelines.
```
Comment (ordirules) [inline paper/paper.md]: What about @mrocklin 's comment? I liked his suggestion. Copying it here again  for convenience:

> We describe the high-level architecture of the project, along with concrete real-world examples in data processing >pipelines in X-ray beamline facilities, particularly ....

Comment (CJ-Wright) [reply paper/paper.md]: That's fair. I've been trying to read more JOSS papers to understand the proper length. Some of the papers seem very short, but others seem a bit longer (although all are under one page ish).

File: paper/paper.md [hunk 9]
```diff
@@ -0,0 +1,57 @@
...
+  - Stream
+  - Dask
+authors:
+ - name: Matthew Rocklin
+   orcid: ?
+   affiliation: 1
+ - name: Christopher J. Wright
+   orcid: 0000-0003-2522-7028
+   affiliation: 2
+ - name: Julien Lhermitte
```
Comment (ordirules) [inline paper/paper.md]: Actually Julien R. Lhermitte as well, thanks!

Comment (danielballan) [reply paper/paper.md]: "Renegade"

Comment (ordirules) [reply paper/paper.md]: "Baller" :-P
(I update myself on urban dictionary quite frequently)

**Comment by mrocklin** [comment]
Broadly speaking I hesitate to push forward on this.  There has been a fair amount of churn recently and this will likely continue into the near future.  It's not clear to me that we should publish any artifact about this at this stage.

**Comment by mrocklin** [comment]
I wonder if, instead, it would make sense to publish things about your downstream code SHED?

**Comment by CJ-Wright** [comment]
That is in the works too, but we thought it would be best to get the base library in the open so that we can cite it for SHED.

**Comment by mrocklin** [comment]
> That is in the works too, but we thought it would be best to get the base library in the open so that we can cite it for SHED.

While I can see the value of a citation I think it's best to wait.  I would like to close this for now.  Are you ok with this?

**Comment by sbillinge** [comment]
I am ok. I don't think we are quite ready.

**Comment by CJ-Wright** [comment]
Ok

**Pull request closed without merging**